### ISSUE TITLE
Intake | RAMP connected to canceled EPs

### ISSUE THREAD

**Issue by pkarman** [department-of-veterans-affairs/caseflow#9476]
When a RAMP EndProductEstablishment is created, it attempts to re-use EPs that have already been created, but ignores their status.

https://github.com/department-of-veterans-affairs/caseflow/blob/8e37e1f5d20628ff66c72e407a0762b17092bec5/app/models/end_product.rb#L113

See https://github.com/department-of-veterans-affairs/caseflow/blob/9f4db2c12b4c36bd0cee8cac1124acd88b53d1ad/app/models/ramp_review.rb#L69

Reported via
https://github.com/department-of-veterans-affairs/appeals-support/issues/3938

## Acceptance criteria

The follow scenario should succeed:

* A RAMP appeal Intake completes
* The EP is canceled
* The same appeal Intake is re-attempted, and a new EP is created (not re-using the existing canceled one).

**Comment by steve-gov** [comment]
@pkarman Do you have an example of a RAMP eligible file number in UAT?

**Comment by pkarman** [comment]
I don't, alas. Any veteran with legacy eligible ratings would qualify afaik, but I don't have a list of those either. Hit me in slack if you need helping tracking one down.

**Comment by Sjones352** [comment]
@pkarman can you please provide me with a veteran file with legacy eligible ratings.

**Comment by Sjones352** [comment]
**Creating the RAMP** 
<img width="1086" alt="Screen Shot 2019-03-13 at 2 28 52 PM" src="https://user-images.githubusercontent.com/23080951/54312631-c9deee80-45ad-11e9-98cd-c92993384264.png">
<img width="1213" alt="Screen Shot 2019-03-13 at 2 29 54 PM" src="https://user-images.githubusercontent.com/23080951/54312632-c9deee80-45ad-11e9-80dd-e932739fc879.png">

<img width="1674" alt="Screen Shot 2019-03-13 at 2 58 08 PM" src="https://user-images.githubusercontent.com/23080951/54312565-a74cd580-45ad-11e9-93e9-ffe0615e9a0e.png">

**Cancel Ep in BGS**
<img width="1680" alt="Screen Shot 2019-03-13 at 4 27 15 PM" src="https://user-images.githubusercontent.com/23080951/54312566-a7e56c00-45ad-11e9-8078-005a1769357a.png">

**Comment by pkarman** [comment]
I'm not sure how a different file number would help. I will investigate 700032765

**Comment by Sjones352** [comment]
### Step 1
So this is the error I get while creating a RAMP
<img width="1308" alt="Screen Shot 2019-03-20 at 1 55 32 PM" src="https://user-images.githubusercontent.com/23080951/54707657-f77ce800-4b17-11e9-9f21-cdfd9e495522.png">

### Step 2
I go to the  RAMP Opt-In Election form to process it in Caseflow.  
<img width="1396" alt="Screen Shot 2019-03-20 at 1 51 06 PM" src="https://user-images.githubusercontent.com/23080951/54707742-2abf7700-4b18-11e9-89a7-ee51c896e776.png">

Not sure what I can do to get past the error.

**Comment by Sjones352** [comment]
I have deleted the Eps for this file number and will check back on the sync job before I try again.
<img width="1680" alt="Screen Shot 2019-03-20 at 5 50 43 PM" src="https://user-images.githubusercontent.com/23080951/54723237-556ff680-4b3d-11e9-82d4-63a3aa675e01.png">

**Comment by Sjones352** [comment]
Created a RAMP opt-in-election and an EP was created. I am going to cancel this BGS, wait for the sync and attempt creating another EP again.

<img width="1326" alt="Screen Shot 2019-03-21 at 1 34 52 PM" src="https://user-images.githubusercontent.com/23080951/54775376-3a0bf680-4be4-11e9-9f76-02b41ac564fd.png">


<img width="1416" alt="Screen Shot 2019-03-21 at 2 16 31 PM" src="https://user-images.githubusercontent.com/23080951/54775357-29f41700-4be4-11e9-8318-2655c606c55c.png">
<img width="1423" alt="Screen Shot 2019-03-21 at 2 17 24 PM" src="https://user-images.githubusercontent.com/23080951/54775358-29f41700-4be4-11e9-8b14-a52572ed5804.png">

**Comment by Sjones352** [comment]
I truly don't know why I am having so much trouble validating this ticket.

<img width="1153" alt="Screen Shot 2019-03-22 at 1 57 45 PM" src="https://user-images.githubusercontent.com/23080951/54843385-9a656b80-4caa-11e9-8197-76905856af49.png">

**Comment by pkarman** [comment]
It worked for me just now. Maybe some timing thing got cleared?

**Comment by Sjones352** [comment]
@pkarman you are right. It was timing issue. 
I have created the EP again. Since I had deleted the previous Ep, this is the only showing in VBMS at the moment.  I am going to wait for the synced_status: "PEND" to go to success before I can close it.
<img width="1680" alt="Screen Shot 2019-03-25 at 3 39 11 PM" src="https://user-images.githubusercontent.com/23080951/54949205-ed425b80-4f14-11e9-8428-516d06ce9656.png">
<img width="1432" alt="Screen Shot 2019-03-25 at 3 39 33 PM" src="https://user-images.githubusercontent.com/23080951/54949206-ed425b80-4f14-11e9-8368-252f9aa2b3e9.png">
<img width="830" alt="Screen Shot 2019-03-25 at 3 41 50 PM" src="https://user-images.githubusercontent.com/23080951/54949207-ed425b80-4f14-11e9-822e-fb183bc60dd2.png">
